### DISCUSSION TITLE
Spore Meeting Agenda @ 2024-01-15

### DISCUSSION THREAD

**Discussion by ashuralyk** [sporeprotocol/spore-scrum#14] [Show and tell]
# Goal

[[previous]](https://github.com/sporeprotocol/spore-scrum/discussions/12)

1. complete integration of co-build in `spore-sdk` _(hard goal)_
2. deploy `spore-demo` on CKB testnet and meets our expectation on its normal operations _(soft goal)_
3. enrich current test cases of `spore-contract`
4. deploy `spore-contract` on CKB mainnet _(hard goal)_

p.s. to start mainnet deployment at least should wait the first goal completed, and it's better to wail the second goal completed if things go well @RongxinZhang 

talks: should we change the implementation of `min_payment`?

# Previous Work

1. apply testing with test cases of `spore-sdk` in the CI workflow of `spore-contract` @ashuralyk 

2. solve bugs and enhance maintenance, test cases and docs of `spore-contract` @ashuralyk 
    - [x] https://github.com/sporeprotocol/spore-contract/pull/32
    - [x] https://github.com/sporeprotocol/spore-contract/pull/42
    - [ ] https://github.com/sporeprotocol/spore-contract/pull/41
    - [ ] https://github.com/sporeprotocol/spore-contract/pull/44

5. deploy `spore-demo` on testnet and switches to match version v2 @Dawn-githup 
    - [x] switch contract address from version v1 to latest version v2
    - [x] collect issues for its successful run https://github.com/sporeprotocol/spore-demo/issues

6. complete integration of co-build into `spore-sdk` and the adaptation to support `spore-demo` @ShookLyngs 
    - [ ] pass CI workflow of this PR https://github.com/sporeprotocol/spore-contract/pull/28
    - [ ] integrate co-build
    - [ ] apply for running `spore-demo`

# Next Plan

1. complete co-build integration and test within at most 3 days
2. successful run `spore-demo` within this week
3. mainnet launch

--------

cc @RongxinZhang @ShookLyngs @Dawn-githup @Flouse

**Comment by ShookLyngs** [comment]
Done:
- Enableed spore-demo for v2-only testing:
  - https://github.com/sporeprotocol/spore-sdk/pull/67
  - https://github.com/sporeprotocol/spore-graphql/pull/1
  - https://github.com/sporeprotocol/spore-demo/tree/feat/v2-contracts-only

WIP:
- Co-build integration in spore-sdk:
  - Testing with legacy ckb default lock: https://www.notion.so/cryptape/Cobuild-only-or-Cobuild-Legacy-a7b8646d18f04c0bbe9f0bd2164617ed#440b62f7417a400a9a5f50693cc240a7
  - Should we test legacy mode only?

**Reply by ashuralyk** [reply]
> Should we test legacy mode only?

currently, yes

**Reply by ShookLyngs** [reply]
Update:
- Co-build integration in spore-sdk: https://github.com/sporeprotocol/spore-sdk/pull/68

**Reply by ashuralyk** [reply]
good, but please do not merge PRs without review and confirmation from others

**Comment by Dawn-githup** [comment]
Done:
  
  1. spore-demo e2e functional test: https://github.com/sporeprotocol/spore-scrum/issues/8#issuecomment-1887142097
  2. spore-dev optimizes details and adds other features：https://github.com/sporeprotocol/spore-devenv/issues/1
  
Next:
  1. Conduct usability testing on spore-demo based on the latest spore-contract.https://github.com/sporeprotocol/spore-scrum/issues/8#issuecomment-1888480013
  2. Build spore-demo e2e automatic testing framework

**Comment by ashuralyk** [comment]
# Progress on Scrum
1. co-build integration has been finished
2. `spore-demo` has run for a while and already fixed some bugs that detected  in `spore-graphql` (demo still has plenty of  bugs)
3. `philosopher stone` has updated its dependencies and run for a while
4. mainnet has not been launched, move to next scrum
